File: HomeLibraryCore/src/main/java/com/smolderingdrake/homelibrarycore/domain/Author.java
package com.smolderingdrake.homelibrarycore.domain;

import lombok.*;
import org.hibernate.validator.constraints.Length;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.Objects;

@Entity(name = "authors")
@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString(exclude = "books")
public class Author {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long idx;

    @Length(max = 50, message = "First name should have up to 50 characters")
    @Column(name = "first_name")
    private String firstName;

    @NotNull(message = "Last name cannot be empty")
    @Length(min = 2, max = 50, message = "Last name should have at least 2 characters and up to 50 characters")
    @Column(name = "last_name")
    private String lastName;

    @Column(name = "description")
    private String description;

    @ManyToMany(mappedBy = "authors")
    private List<Book> books;

    public boolean equals(Author author) {
        if (this == author) return true;
        if (author == null) return false;
        return firstName.equals(author.firstName)
                && lastName.equals(author.lastName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstName, lastName);
    }
}

File: HomeLibraryWeb/src/app/author.ts
import {Book} from './book';

export class Author {
  idx: bigint;
  firstName: string;
  lastName: string;
  books: Book[] = [];
  description: string;

  convert(singleAuthorArray) {
    this.idx = singleAuthorArray[0];
    this.firstName = singleAuthorArray[1];
    this.lastName = singleAuthorArray[2];
    this.description = singleAuthorArray[3];
    this.books = singleAuthorArray[4];
  }
}
